### ISSUE TITLE
Transformation API (transform the generated schema during build step)

### ISSUE THREAD

**Issue by Weakky** [graphql-nexus/nexus-plugin-prisma#110]
## Description

Once #86 lands, we will be in a better situation to pass parameters to nexus-prisma-generate. We think it might add a lot of value to be able to transform the generated schema during the build step.

That would not only reduce the runtime overhead, but also provide type-safety over the transformed schema and probably unlock some transformation that wouldn't be possible/be harder to implement at runtime.

## Use-cases

- #90 
- #95 
- Rename args
- Renamed type names (input, object, enum ...)
- Disable connection types or any other types (like Subscription if not needed)

Do you have any use-cases in mind? Let's use this thread as a place to discuss these 🙌

**Comment by jasonkuhrt** [comment]
@Weakky can we recycle this into a P2 discussion or should we open a new issue for that?

**Comment by Weakky** [comment]
Interesting use-case from @Fonz001
👉 Have all nested inputs only allow creation
![image](https://user-images.githubusercontent.com/25233379/65892992-857b3a80-e3a7-11e9-8722-7b465a5293ad.png)
> (Screenshotted from Slack with his permission)

@jasonkuhrt We can recycle this into a P2 discussion 👍

**Comment by PatrickVanRietschoten** [comment]
Agreed to the above! It would be awesome to have our Input types compatible with the Prisma input types. 

In that case - we have a standardization across all our input types AND we can pass the input directly to Prisma :-)
